csrc/env.h

```diff
@@ -61,16 +61,22 @@ class java_ex {
 
     public:
         java_ex(jthrowable exception) COLD
-            : m_java_exception(exception), m_java_classname(nullptr), m_message(), m_message_cstr("")
+            : m_java_exception(exception), m_java_classname(nullptr)
         { }
 
         java_ex(const char *java_classname, const char *message) COLD
-            : m_java_exception(nullptr), m_java_classname(java_classname), m_message(), m_message_cstr(message)
-        { capture_trace(); }
+            : m_java_exception(nullptr), m_java_classname(java_classname) {
+
+            capture_trace();
+
+            if (message != NULL) {
+                size_t amount_to_copy = strnlen(message, (sizeof(m_message_cstr) - 1));
+                std::strncpy(m_message_cstr, message, amount_to_copy);
```

Comment:
While I believe that we should be doing this all through `m_message` rather than `m_message_cstr`, why is this not the simpler:
```c++
std::strncpy(m_message_cstr, message, sizeof(m_message_cstr) - 1);
```

Reply:
This was a misunderstanding on my part of the Eclipse IDE `strncpy` documentation that states `This function is similar to strcpy but always copies exactly size characters into to.` which made it sound like if `sizeof(m_message_cstr) - 1` was passed in that strncpy might read past the end of the from buffer. Other documentation sources are more clear that this isn't the case.